Retour des npc et dialogues & restructuration du code

antoinech2 opened the pull request:
- Permet de rendre à nouveau compatible les NPC et les dialogues avec le nouveau système de map, avec une gestion des npc par map.
- Restructuration des importations de tous les fichiers, désormais, la structure suit ce que l'on avait décidé. Chaque class est bien séparée et n'appelle que des éléments qui sont logique avec elle
- Restructuration des NPC (compatible par map), des dialogues, et du déplacement du joueur

Note : L'héritage du joueur et des NPC n'a pas encore été mis en place. Le fichier entity est mis avec un "-" pour signifier qu'il n'est pas encore utilisé

antoinech2 committed:
Ajout de la compatibilité des NPC
Les dialogues ne sont pas fonctionnel
Redesign du fichier npc

antoinech2 committed:
Réintroduction des dialogues avec restructuration
L'affichage du texte n'est pas fonctionnel

antoinech2 committed:
Restructuration des dialogues
Les dialogues sont maintenant fonctionnels

antoinech2 committed:
Connventionnalisation

antoinech2 committed:
Restructuration du déplacement du joueur et conv
Préparation au support des textures en diagonales (#12)
Résolution de #14
Avancée sur #5

antoinech2 committed:
Ajout du support de multiples dialogues pour un npc

fmkr-project commented:
Spam dans la console, peut être gênant si le jeu tourne pendant longtemps

https://user-images.githubusercontent.com/91734800/145713269-7498cf7a-914c-4e04-949e-bb3d64c7aff9.mp4

antoinech2 commented:
> Spam dans la console, peut être gênant si le jeu tourne pendant longtemps
> 
>  2021-12-12-13-55-38.mp4

Ce problème vient de l'utilisation de IDLE qui ne supporte pas bien l'affichage console. Il n'y a aucun problème en exécutant directement avec python. Je vais quand même supprimer le log

fmkr-project commented:
Après un peu de testing le dialogue lague lorsque du texte est imprimé dans la console

fmkr-project commented:
ok

antoinech2 committed:
Suppression du log avec le temps écoulé

fmkr-project reviewed (changes requested):
Plus aucun warp excepté celui de l'escalier principal ne marche 

https://user-images.githubusercontent.com/91734800/145714044-29539391-9eeb-42c0-8251-085611a24985.mp4

antoinech2 committed:
Hotfix : Potail ne fonctionne plus.
Le fix est très mauvais, mais une restructuration complète de la classe 
Map est nécessaire pour régler convenablement le problème

fmkr-project committed:
Implémentation #22. Fix #11.

fmkr-project committed:
Hotfix barre d'espace

fmkr-project committed:
Hotfix #11

antoinech2 committed:
On enlève le tiret pour faire plaisir à Matéo
